urlb57eb64056_jacadenac_ISI_Othello_UN_tgz-pJ8-othello_OthelloJ8/src/logica/ia/algoritmosGeneticos/Estrategia.java

```diff
@@ -0,0 +1,435 @@
+		// Create XML document from generated tree
+		XMLDocumentBuilder docbuilder = new XMLDocumentBuilder();
+		Document xmlDoc = (Document) docbuilder.buildDocument(doc2);
+		XMLManager.writeFile(xmlDoc, new File("PoblacionEstrategia.xml"));
+		
+	}
+	
+	// -------------------------------------------------------------------
+	// Este metodo permite obtiene la población a partir de un archivo
+	// -------------------------------------------------------------------
+	
+	public static Genotype obtenerPoblacion(Configuration conf) throws InvalidConfigurationException{
+		int[][] estrategiasCompetidoras = leerPoblacion();//obtenerCompetidores();
+		Chromosome[] chromosomas= new Chromosome[cantidadCompetidores()];
+		for(int i=0 ; i<chromosomas.length ; i++){
+			Gene[] genes = new Gene[4];
+			for(int j=0 ; j<genes.length ; j++){
+				genes[j] = new IntegerGene(conf, 0, RANGO_GEN);
+				genes[j].setAllele((Integer)(estrategiasCompetidoras[i][j]));
+			}		
+			chromosomas[i] = new Chromosome(conf, genes);	
+			/**/int[] cromosoma = obtenerValoresCromosoma(chromosomas[i]);
+			/**///System.out.println(cromosoma[0]+","+cromosoma[1]+","+cromosoma[2]+","+cromosoma[3]);
+		}
+		Population population = new Population(conf,chromosomas);
+		return new Genotype(conf,population);
+	}
+	
+	public static int cantidadCompetidores(){
+		return leerPoblacion().length;
+	}
+	
+	private static int[] obtenerValoresCromosoma(IChromosome cromosoma) {
+		int[] valores = new int[4];
+		for (int i=0; i<4; i++){
+			valores[i]=getPesoDeGen(cromosoma, i);
+		}
+		return valores;
+	}
+	
+	private static int getPesoDeGen(IChromosome cromosoma, int i) {
+		Integer peso = (Integer) cromosoma.getGene(i).getAllele();
+		return peso;
+	}
+	
+	private static int[][] leerPoblacion() {
+		    int[][] poblacion = new int[17][5];  //tamańo población inicial -> 15
+			JFileChooser fileChooser = new JFileChooser();
+			//String nombreFichero = "poblacionInicial.txt";
+			String nombreFichero = "ultimaGeneracion.txt";
+			File fichero = new File(directorio.getPath(),nombreFichero);
+			fileChooser.setCurrentDirectory(fichero);		
+			try {
+				BufferedReader reader = new BufferedReader(new FileReader(fichero));
+				String linea = reader.readLine();
+				int i = 0;
+				while (linea != null) {
+					String[] dimension = linea.split ("-");
+					poblacion[i][4] = Integer.parseInt(dimension[0]);
+					dimension = dimension[1].split(",");
+					for (int j=0 ; j<dimension.length ; j++) {				
+						poblacion[i][j]=Integer.parseInt(dimension[j]);
+					}
+					linea = reader.readLine();
+					i++;
+				}
+				reader.close();
+			} catch (FileNotFoundException e1) {
+				e1.printStackTrace();
+			} catch (IOException e) {
+				e.printStackTrace();
+			}
+			return poblacion;
+	}
+	
+	public static String poblacionToString(IChromosome[] cromosomas){
+		String poblacion = "";
+		String nombreFichero = "ultimaGeneracion.txt";
+		if (!directorio.exists()) {
+            directorio.mkdirs();
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

Confirm that `mkdirs` was successful. Handle any failures that happened. If you fail silently, then you might encounter errors that are harder to debug. See the following diffs on GitHub that make this code change: [diff1](https://github.com/qunarcorp/qmq/commit/12f1d9d6a57a02bc2415cb2af9f4388a77b12c02#diff-6b67be6e17f24fdeddb7c725ac8ec5600d6e63be0bdca0963a83d530e6d55e35L36), [diff2](https://github.com/apache/jmeter/commit/1ec61206779b3f84df18516682b4791bf015178f#diff-cd74304763c097fbc085ac2bc952b46ab3843ac825cacacad89e8064296e955aL330)

Learn more: [File.mkdirs in the Java Platform, Standard Edition 8 API Specification](https://docs.oracle.com/javase/8/docs/api/java/io/File.html#mkdirs--)